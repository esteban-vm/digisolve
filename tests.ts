import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
