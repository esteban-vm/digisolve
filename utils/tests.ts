import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
