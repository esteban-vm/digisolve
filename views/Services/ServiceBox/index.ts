export { default, type ServiceProps } from './ServiceBox'
