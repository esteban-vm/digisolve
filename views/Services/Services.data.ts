import type { ServiceProps } from './Service'

const description = 'Ipsam reiciendis eligendi est qui itaque.'

const services: ServiceProps[] = [
  {
    id: '99a36a03-7210-44c9-b031-f0e2acec54e5',
    title: 'branding',
    icon: 'tags',
    description,
  },
  {
    id: '82879f1c-49d5-4b10-b419-043556c4aa3b',
    title: 'content',
    icon: 'pen-clip',
    description,
  },
  {
    id: '7ec3da85-2944-4097-b198-f1fc7a0863a7',
    title: 'marketing',
    icon: 'filter-circle-dollar',
    description,
  },
  {
    id: 'fba92296-7d8b-4150-b10d-2f1a90041c65',
    title: 'web design',
    icon: 'code',
    description,
  },
]

export default services
