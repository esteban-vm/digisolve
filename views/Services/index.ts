export { default as Services, services } from './Services'
