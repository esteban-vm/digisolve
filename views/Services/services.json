[
  { "title": "branding", "icon": "tags" },
  { "title": "content", "icon": "pen-alt" },
  { "title": "marketing", "icon": "funnel-dollar" },
  { "title": "web design", "icon": "laptop-code" }
]
