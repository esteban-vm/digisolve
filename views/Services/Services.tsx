import type { FunctionalComponent } from '@/types'
import { ServiceItem, type ServiceProps } from '@/components'
import { faTags, faPenAlt, faFunnelDollar, faLaptopCode } from '@fortawesome/free-solid-svg-icons'
import { styled } from '@/utils/styles'

export const services: ServiceProps[] = [
  { heading: 'Branding', icon: faTags },
  { heading: 'Content', icon: faPenAlt },
  { heading: 'Marketing', icon: faFunnelDollar },
  { heading: 'Web Design', icon: faLaptopCode },
]

const ServiceComponent: FunctionalComponent = (props) => {
  return (
    <section {...props}>
      <div className='row'>
        <h2 className='section_heading'>Our Services</h2>
        <h3 className='section_subheading'>Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.</h3>
      </div>
      <div className='row'>
        {services.map((service, index) => (
          <ServiceItem key={index} {...service} role='listitem' />
        ))}
      </div>
    </section>
  )
}

const Services = styled(ServiceComponent)`
  text-align: center;
  background-color: var(--color-light);
`

export default Services
