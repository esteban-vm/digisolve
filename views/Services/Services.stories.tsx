import type { Meta } from '@/stories'
import Services from './Services'

export default {
  component: Services,
  title: 'Views/Services',
  parameters: {
    backgrounds: {
      disable: true,
    },
  },
} as Meta<typeof Services>

export const View = () => <Services />
