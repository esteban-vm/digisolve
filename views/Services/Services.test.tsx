import { render, cleanup, screen, create } from '@/tests'
import services from './services.json'
import Services from './Services'

describe('ðŸ§ª SERVICES:', () => {
  describe('display tests:', () => {
    /** Section title */
    const name = /^our services$/i

    beforeEach(() => {
      render(<Services />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const section = screen.getByRole('region', { name })
      expect(section).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the heading', () => {
        const heading = screen.getByRole('heading', { name, level: 2 })
        expect(heading).toBeInTheDocument()
        expect(heading).toBeVisible()
      })

      it('the subheading', () => {
        const subheading = screen.getByRole('heading', { level: 3 })
        expect(subheading).toBeInTheDocument()
        expect(subheading).toBeVisible()
      })

      it('the services', () => {
        const serviceElements = screen.getAllByRole('article')
        expect(serviceElements).toHaveLength(services.length)

        for (const service of serviceElements) {
          expect(service).toBeInTheDocument()
          expect(service).toBeVisible()
        }
      })
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<Services />)
      expect(tree).toMatchSnapshot()
    })
  })
})
