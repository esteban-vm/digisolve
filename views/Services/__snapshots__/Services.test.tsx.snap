// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª SERVICES: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: center;
  background-color: var(--color-light);
}

.emotion-1 {
  padding: 1% 1.5%;
}

.emotion-1 .service__icon {
  width: 3rem;
  height: 3rem;
  margin-bottom: 30px;
  color: var(--color-secondary);
}

.emotion-1 .service__heading {
  font-size: 150%;
  margin-bottom: 30px;
  font-weight: 400;
  text-transform: capitalize;
}

.emotion-1 .service__paragraph {
  color: var(--color-grey);
  font-size: 90%;
  margin-bottom: 20px;
}

.emotion-1 .service__link {
  color: var(--color-primary);
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
}

.emotion-1 .service__link:hover {
  opacity: 0.7;
}

<section
  aria-labelledby=":r0:"
  className="emotion-0"
>
  <div
    className="row"
    role="presentation"
  >
    <h2
      className="section__heading"
      id=":r0:"
    >
      Our services
    </h2>
    <h3
      className="section__subheading"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.
    </h3>
  </div>
  <div
    className="row"
    role="presentation"
  >
    <div
      className="col span_1_of_4"
      role="presentation"
    >
      <article
        className="emotion-1"
      >
        <svg
          aria-labelledby="svg-inline--fa-title-t98cMRBgXiql"
          className="svg-inline--fa fa-tags service__icon"
          data-icon="tags"
          data-prefix="fas"
          role="img"
          style={{}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-t98cMRBgXiql"
            style={{}}
          >
            branding
          </title>
          <path
            d="M345 39.1L472.8 168.4c52.4 53 52.4 138.2 0 191.2L360.8 472.9c-9.3 9.4-24.5 9.5-33.9 .2s-9.5-24.5-.2-33.9L438.6 325.9c33.9-34.3 33.9-89.4 0-123.7L310.9 72.9c-9.3-9.4-9.2-24.6 .2-33.9s24.6-9.2 33.9 .2zM0 229.5V80C0 53.5 21.5 32 48 32H197.5c17 0 33.3 6.7 45.3 18.7l168 168c25 25 25 65.5 0 90.5L277.3 442.7c-25 25-65.5 25-90.5 0l-168-168C6.7 262.7 0 246.5 0 229.5zM144 144a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <h4
          className="service__heading"
        >
          branding
        </h4>
        <p
          className="service__paragraph"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem, debitis illum expedita nulla quae consectetur.
        </p>
        <a
          className="service__link"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
    <div
      className="col span_1_of_4"
      role="presentation"
    >
      <article
        className="emotion-1"
      >
        <svg
          aria-labelledby="svg-inline--fa-title-4AGIi8SolaNK"
          className="svg-inline--fa fa-pen-clip service__icon"
          data-icon="pen-clip"
          data-prefix="fas"
          role="img"
          style={{}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-4AGIi8SolaNK"
            style={{}}
          >
            content
          </title>
          <path
            d="M453.3 19.3l39.4 39.4c25 25 25 65.5 0 90.5l-52.1 52.1 0 0-1-1 0 0-16-16-96-96-17-17 52.1-52.1c25-25 65.5-25 90.5 0zM241 114.9c-9.4-9.4-24.6-9.4-33.9 0L105 217c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L173.1 81c28.1-28.1 73.7-28.1 101.8 0L288 94.1l17 17 96 96 16 16 1 1-17 17L229.5 412.5c-48 48-109.2 80.8-175.8 94.1l-25 5c-7.9 1.6-16-.9-21.7-6.6s-8.1-13.8-6.6-21.7l5-25c13.3-66.6 46.1-127.8 94.1-175.8L254.1 128 241 114.9z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <h4
          className="service__heading"
        >
          content
        </h4>
        <p
          className="service__paragraph"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem, debitis illum expedita nulla quae consectetur.
        </p>
        <a
          className="service__link"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
    <div
      className="col span_1_of_4"
      role="presentation"
    >
      <article
        className="emotion-1"
      >
        <svg
          aria-labelledby="svg-inline--fa-title-FkHA9gs5XZ0d"
          className="svg-inline--fa fa-filter-circle-dollar service__icon"
          data-icon="filter-circle-dollar"
          data-prefix="fas"
          role="img"
          style={{}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-FkHA9gs5XZ0d"
            style={{}}
          >
            marketing
          </title>
          <path
            d="M3.9 22.9C10.5 8.9 24.5 0 40 0H472c15.5 0 29.5 8.9 36.1 22.9s4.6 30.5-5.2 42.5L396.4 195.6C316.2 212.1 256 283 256 368c0 27.4 6.3 53.4 17.5 76.5c-1.6-.8-3.2-1.8-4.7-2.9l-64-48c-8.1-6-12.8-15.5-12.8-25.6V288.9L9 65.3C-.7 53.4-2.8 36.8 3.9 22.9zM576 368c0 79.5-64.5 144-144 144s-144-64.5-144-144s64.5-144 144-144s144 64.5 144 144zM408.8 335.4c.6-.9 1.8-2.1 4.2-3.4c5.1-2.7 12.5-4.1 18.7-4c8.2 .1 17.1 1.8 26.4 4.1c8.6 2.1 17.3-3.1 19.4-11.7s-3.1-17.3-11.7-19.4c-5.6-1.4-11.6-2.7-17.9-3.7V288c0-8.8-7.2-16-16-16s-16 7.2-16 16v9.5c-6.1 1.2-12.3 3.2-18 6.3c-11.8 6.3-23 18.4-21.8 37.2c1 16 11.7 25.3 21.6 30.7c8.8 4.7 19.7 7.8 28.6 10.3l1.8 .5c10.3 2.9 17.9 5.2 23.2 8.3c4.5 2.7 4.7 4.2 4.7 5.6c.1 2.4-.5 3.7-1 4.5c-.6 1-1.8 2.2-4 3.3c-4.7 2.5-11.8 3.8-18.5 3.6c-9.5-.3-18.5-3.1-29.9-6.8c-1.9-.6-3.8-1.2-5.8-1.8c-8.4-2.6-17.4 2.1-20 10.5s2.1 17.4 10.5 20c1.6 .5 3.3 1 5 1.6l0 0 0 0c7 2.3 15.1 4.8 23.7 6.6v11.4c0 8.8 7.2 16 16 16s16-7.2 16-16V438.7c6.2-1.1 12.5-3.1 18.3-6.2c12.1-6.5 22.3-18.7 21.7-36.9c-.5-16.2-10.3-26.3-20.5-32.3c-9.4-5.6-21.2-8.9-30.5-11.5l-.2 0c-10.4-2.9-18.3-5.2-23.9-8.2c-4.8-2.6-4.8-4-4.8-4.5l0-.1c-.1-1.9 .3-2.9 .8-3.6z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <h4
          className="service__heading"
        >
          marketing
        </h4>
        <p
          className="service__paragraph"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem, debitis illum expedita nulla quae consectetur.
        </p>
        <a
          className="service__link"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
    <div
      className="col span_1_of_4"
      role="presentation"
    >
      <article
        className="emotion-1"
      >
        <svg
          aria-labelledby="svg-inline--fa-title-hr4I8a319nrr"
          className="svg-inline--fa fa-laptop-code service__icon"
          data-icon="laptop-code"
          data-prefix="fas"
          role="img"
          style={{}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-hr4I8a319nrr"
            style={{}}
          >
            web design
          </title>
          <path
            d="M64 96c0-35.3 28.7-64 64-64H512c35.3 0 64 28.7 64 64V352H512V96H128V352H64V96zM0 403.2C0 392.6 8.6 384 19.2 384H620.8c10.6 0 19.2 8.6 19.2 19.2c0 42.4-34.4 76.8-76.8 76.8H76.8C34.4 480 0 445.6 0 403.2zM281 209l-31 31 31 31c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-48-48c-9.4-9.4-9.4-24.6 0-33.9l48-48c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9zM393 175l48 48c9.4 9.4 9.4 24.6 0 33.9l-48 48c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l31-31-31-31c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <h4
          className="service__heading"
        >
          web design
        </h4>
        <p
          className="service__paragraph"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Rem, debitis illum expedita nulla quae consectetur.
        </p>
        <a
          className="service__link"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Read more
        </a>
      </article>
    </div>
  </div>
</section>
`;
